docs/next/guides/building-and-testing/plugins.md

```diff
@@ -17,228 +15,242 @@ tags:
+    // plugin hooks and clearing all the plugin properties.
     super.destroy();
   }
 }
-
-export {InternalPluginSkeleton};
-
-// You need to register your plugin in order to use it within Handsontable.
-registerPlugin('internalPluginSkeleton', InternalPluginSkeleton);
-
 ```
 
-### External plugin
-
-Similarly to the above, this template is [hosted on GitHub](https://github.com/handsontable/handsontable-skeleton/tree/master/plugins/external).
-
-```js
-/**
- * @plugin External plugin skeleton.
- * Note: keep in mind, that Handsontable instance creates one instance of the plugin class.
- *
- * @param hotInstance
- * @constructor
- */
-function ExternalPluginSkeleton(hotInstance) {
-
-  // Call the BasePlugin constructor.
-  Handsontable.plugins.BasePlugin.call(this, hotInstance);
+### 3. Register CustomPlugin
+Last but not least, you need to register the plugin. This is the only way to use plugins in Handsontable.
+There are two ways to do that.
 
-  this._superClass = Handsontable.plugins.BasePlugin;
+- You can define a static getter named `PLUGIN_KEY` that the `registerPlugin` utility uses as the plugin's alias. Usage example you can find in the above code snippet.
+  ```js
+  // You need to register your plugin in order to use it within Handsontable.
+  registerPlugin(CustomPlugin);
+  ```
+- You can also use the custom alias. Put a string in the first argument. The registerer will use that string as an alias instead of the `PLUGIN_KEY` getter from `CustomPlugin`.
+  ```js
+  registerPlugin('CustomAlias', CustomPlugin);
+  ```
 
-  // Initialize all your public properties in the class' constructor.
-  /**
-   * yourProperty description.
-   *
-   * @type {String}
-   */
-  this.yourProperty = '';
-  /**
-   * anotherProperty description.
-   * @type {Array}
-   */
-  this.anotherProperty = [];
-}
+### 4. Use plugin in Handsontable
+Accordingly, to the supported options for the plugin, you might control it, passing in its options the boolean or object.
+The following snippet presents all possible approaches:
 
-// Inherit the BasePlugin prototype.
-ExternalPluginSkeleton.prototype = Object.create(Handsontable.plugins.BasePlugin.prototype, {
-  constructor: {
-    writable: true,
-    configurable: true,
-    value: ExternalPluginSkeleton
+```js
+import Handsontable from 'handsontable';
+import { CustomPlugin } from './customPlugin';
+
+const hotInstance = new Handsontable(container, {
+  // You can enable plugin by passing `true` to use default options.
+  [CustomPlugin.PLUGIN_KEY]: true,
+  // You can also enable the plugin by passing an object with options.
+  [CustomPlugin.PLUGIN_KEY]: {
+    msg: 'user-defined message',
   },
+  // You can also initialize the plugin without enabling it at the beginning.
+  [CustomPlugin.PLUGIN_KEY]: false,
 });
+```
+
+### 5. Get a reference to the plugin's instance
+To use the plugin's API you may use [`getPlugin`](../api/core/#getplugin) method to get a reference to its instance.
```

Comment:
To use the plugin's API, call the [\`getPlugin\`](../api/core/#getplugin) method to get a reference to the plugin's instance.